### DISCUSSION TITLE
Troubles making correct Model validation

### DISCUSSION THREAD

**Discussion by stefanodvx** [pydantic/pydantic#10611] [Question]
Hi guys!
Let me explain. I got a main model:
```
class Member(BaseModel):
    ...
```
And the other models
```
class Owner(BaseModel):
    ...
class Admin(BaseModel):
    ...
```
I'd like to use `Member.model_validate(data)` and get the corrensponding model back. Every model (Owner, Admin, ecc...) ofcourse have unique fields so that can be discriminated. I tried using `RootModel` and a `root` field with `Union[Owner, Admin, ...]`. This worked pretty well, but the problem is that when using `Member.model_validate(data)`, I got a `Member` object back, with the `root` field pointing to the correct model. Instead I'd like to dinamically, like the RootModel, directly get the corrensponding model. And eventually make `Member` model inherit type hinting from it's "children" (Owner, Administrator)

Thank you for your time. <3

**Comment by Viicos** [comment]
You might want to take a look at https://docs.pydantic.dev/latest/concepts/unions/#discriminated-unions.

**Reply by stefanodvx** [reply]
This is what I was thinking for, but I don't think I can use it how I want it to be. Using Unions, I should add another field to the `Member` class and access it. Instead I want `Member.model_validate(...)` to directly return the model I want (Owner, Admin, ecc...)
